vsphere/resource_vsphere_license.go

```diff
@@ -0,0 +1,287 @@
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/license"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+var (
+	// ErrNoSuchKeyFound is an error primarily thrown by the Read method of the resource.
+	// The error doesn't display the key itself for security reasons.
+	ErrNoSuchKeyFound = errors.New("The key was not found")
+	// ErrKeyCannotBeDeleted is an error which occurs when a key that is used by VMs is
+	// being removed
+	ErrKeyCannotBeDeleted = errors.New("The key wasn't deleted")
+)
+
+func resourceVSphereLicense() *schema.Resource {
+
+	return &schema.Resource{
+
+		SchemaVersion: 1,
+
+		Create: resourceVSphereLicenseCreate,
+		Read:   resourceVSphereLicenseRead,
+		Update: resourceVSphereLicenseUpdate,
+		Delete: resourceVSphereLicenseDelete,
+
+		Schema: map[string]*schema.Schema{
+			"license_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"label": &schema.Schema{
+				Type:     schema.TypeList,
```

Comment:
This should be a `TypeSet` so that conflicting entries can't be added.

Also, you will need a hashing function as well when converting it to a set - the stock [`HashResource` function](https://github.com/hashicorp/terraform/blob/master/helper/schema/set.go#L23) should help too.